### PR TITLE
[v2.11.0] Removal of catalog v1

### PR DESCRIPTION
**Pull request by rohitsakala**
Fixes https://github.com/rancher/rancher/issues/39525

Please check the issue for more details.

### Summary 

- Catalogv1 is deprecated in previous versions. So removing the codebase related to it now in Rancher 2.11.0. 
- Please ignore the shepherd failures in the CI. It will be fixed before merging.

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -118,8 +114,7 @@ RUN mkdir -p /var/lib/rancher-data/local-catalogs/system-library && \
     git clone -b $CATTLE_PARTNER_CHART_DEFAULT_BRANCH --depth 1 https://git.rancher.io/partner-charts /var/lib/rancher-data/local-catalogs/v2/rancher-partner-charts/8f17acdce9bffd6e05a58a3798840e408c4ea71783381ecd2e9af30baad65974 && \
     git clone -b $CATTLE_RKE2_CHART_DEFAULT_BRANCH --depth 1 https://git.rancher.io/rke2-charts /var/lib/rancher-data/local-catalogs/v2/rancher-rke2-charts/675f1b63a0a83905972dcab2794479ed599a6f41b86cd6193d69472d0fa889c9 && \
     # Revert the previous change in git.rancher.io from .gitconfig
-    rm "${HOME}/.gitconfig" && \
-    git clone -b master --depth 1 https://github.com/rancher/charts /var/lib/rancher-data/local-catalogs/library
```
Comment (pjbgf) [inline package/Dockerfile:122]: Similar changes are required on 
`/tests/v2/codecoverage/package/Dockerfile`.

File: package/jailer.sh [hunk 2]
```diff
@@ -57,21 +57,6 @@ cp -r -l /opt/drivers/management-state/bin /opt/jail/$NAME/var/lib/rancher/manag
...
-cp -l /usr/bin/helm_v3 /opt/jail/$NAME/usr/bin
-
-# Hard link tiller into the jail
-cp -l /usr/bin/rancher-tiller /opt/jail/$NAME/usr/bin
-
-# Hard link kustomize into the jail
-cp -l /usr/bin/kustomize /opt/jail/$NAME/usr/bin
-
-# Hard link custom kustomize script into jail
-cp -l /usr/bin/kustomize.sh /opt/jail/$NAME
```
Comment (pjbgf) [inline package/jailer.sh:73]: This is potentially used by Node drivers. I would probably leave `kustomize{,.sh}` and `helm_v3`.

In doubt, it may be worth reaching out to hostbusters to confirm.

Comment (jiaqiluo) [reply package/jailer.sh:73]: We shoud keep the `kustomize`, `kustomize.sh`, and `helm_v3` used for managing Catalog App. Rancher runs the operation in a jail.

**Comment by pjbgf** [comment]
We may want to remove the testing code touching `git.rancher.io/system-charts`: 

image

**Comment by pjbgf** [comment]
Shall we also remove:
- `pkg/controllers/managementuserlegacy`
- `pkg/controllers/managementlegacy`
- [features.Legacy](https://github.com/rancher/rancher/blob/main/pkg/features/feature.go#L80-L85)

File: go.mod [hunk 3]
```diff
@@ -14,6 +14,7 @@ replace (
 
 	github.com/rancher/rancher/pkg/apis => ./pkg/apis
 	github.com/rancher/rancher/pkg/client => ./pkg/client
+	github.com/rancher/shepherd => github.com/rohitsakala/shepherd v0.0.0-20241029171404-0c24dd625fb4
```
Comment (tashima42) [inline go.mod:17]: Maybe this is a replacement that you forgot to remove

**Review by tashima42 — APPROVED** [review]
GH actions code looks fine to me

**Comment by diogoasouza** [comment]
We also need to remove the roles related to catalogv1 that can be assigned when creating/adding a new user to the cluster.
Example of one, there's more:
image

**Review by diogoasouza — APPROVED** [review]

File: pkg/controllers/management/auth/prtb_handler.go
Comment (JonCrowther) [inline pkg/controllers/management/auth/prtb_handler.go:1]: The file `crtb_handler.go` in this package also has several references to `catalogtemplates` and other CRDs. Could you take a look and remove references there?

**Review by JonCrowther — COMMENTED** [review]
From the RBAC side this looks pretty good. Just missed 1 file I think `pkg/controllers/management/auth/prtb_handler.go`

File: pkg/apis/project.cattle.io/v3/types.go [hunk 4]
```diff
@@ -167,3 +167,8 @@ type WorkloadMetric struct {
 	Path   string `json:"path,omitempty"`
 	Schema string `json:"schema,omitempty" norman:"type=enum,options=HTTP|HTTPS"`
 }
+
+type RollbackRevision struct {
```
Comment (jiaqiluo) [inline pkg/apis/project.cattle.io/v3/types.go:171]: I wonder where the new type is from?

File: pkg/controllers/management/auth/crtb_handler.go
Comment (jiaqiluo) [inline pkg/controllers/management/auth/crtb_handler.go:1]: Some of the fields that were removed don’t appear to be related to catalog v1. Could you confirm that it’s safe and expected to drop them?

File: pkg/controllers/managementuserlegacy/systemimage/systemimage.go [hunk 5]
```diff
@@ -1,122 +0,0 @@
...
-	projects       v3.ProjectInterface
-	systemServices map[string]SystemService
-}
-
-func (s *Syncer) SyncProject(key string, obj *v3.Project) (runtime.Object, error) {
-	if obj == nil || obj.DeletionTimestamp != nil {
-		return nil, nil
-	}
-	return obj, s.Sync()
-}
```
Comment (jiaqiluo) [inline pkg/controllers/managementuserlegacy/systemimage/systemimage.go:30]: Is it expected to remove this function which seems to be related to v3 project? Or should we remove only the logic related to catalog?

File: pkg/controllers/management/rbac/rbac.go [hunk 6]
```diff
@@ -20,10 +20,6 @@ const (
 	OwnerAccess                     = "owner"
 	MemberAccess                    = "member"
 	ReadOnlyAccess                  = "read-only"
-	MultiClusterAppResource         = "multiclusterapps"
-	MultiClusterAppRevisionResource = "multiclusterapprevisions"
-	GlobalDNSResource               = "globaldnses"
-	GlobalDNSProviderResource       = "globaldnsproviders"
```
Comment (jiaqiluo) [inline pkg/controllers/management/rbac/rbac.go:26]: Those two don't seem related to catalog v1. Can you confirm if they are deprecated?

File: pkg/data/management/role_data.go
Comment (jiaqiluo) [inline pkg/data/management/role_data.go:1]: Any rules related to `"apps.project.cattle.io`  and `apprevisions.project.cattle.io` should be removed from the file.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: pkg/controllers/management/auth/crtb_handler.go
Comment (JonCrowther) [reply pkg/controllers/management/auth/crtb_handler.go:1]: I can confirm that those should be removed. They were missed when alerting v1 was deprecated.

File: pkg/controllers/managementuserlegacy/systemimage/systemimage.go (hunk 5 above)
Comment (jiaqiluo) [reply pkg/controllers/managementuserlegacy/systemimage/systemimage.go:30]: yeah, that is right - the whole systemimage is useless.

File: pkg/data/management/role_data.go
Comment (jiaqiluo) [reply pkg/data/management/role_data.go:1]: There are still rules that start with `addRule().apiGroups("project.cattle.io").resources("apps")` or `addRule().apiGroups("project.cattle.io").resources("apprevisions")` in the file. Can you remove all occurrences of them?

File: pkg/controllers/management/auth/crtb_handler.go [hunk 7]
```diff
@@ -68,10 +68,7 @@ const (
 
 var clusterManagementPlaneResources = map[string]string{
 	"clusterscans":                "management.cattle.io",
-	"catalogtemplates":            "management.cattle.io",
-	"catalogtemplateversions":     "management.cattle.io",
-	"clustercatalogs":             "management.cattle.io",
-	"clustermonitorgraphs":        "management.cattle.io",
+	"clusterloggings":             "management.cattle.io",
```
Comment (JonCrowther) [inline pkg/controllers/management/auth/crtb_handler.go:71]: Guessing this is a rebase/merge hiccup. I removed this here https://github.com/rancher/rancher/pull/47997 so could you remove `clusterloggings`?

File: pkg/controllers/management/auth/prtb_handler.go [hunk 8]
```diff
@@ -24,21 +24,12 @@ const (
...
 
 var projectManagementPlaneResources = map[string]string{
-	"apps":                        "project.cattle.io",
-	"apprevisions":                "project.cattle.io",
-	"catalogtemplates":            "management.cattle.io",
-	"catalogtemplateversions":     "management.cattle.io",
 	"sourcecodeproviderconfigs":   "project.cattle.io",
-	"projectcatalogs":             "management.cattle.io",
-	"projectmonitorgraphs":        "management.cattle.io",
+	"projectloggings":             "management.cattle.io",
```
Comment (JonCrowther) [inline pkg/controllers/management/auth/prtb_handler.go:28]: Could you remove `projectloggings`. My PR removing it https://github.com/rancher/rancher/pull/47997

File: pkg/controllers/management/rbac/legacy_ownerrefs_cleaner.go
Comment (JonCrowther) [inline pkg/controllers/management/rbac/legacy_ownerrefs_cleaner.go:1]: Think this accidentally got included from here https://github.com/rancher/rancher/pull/42847

File: pkg/crds/names.go [hunk 9]
```diff
@@ -182,22 +173,17 @@ func UIPluginsCRD() []string {
...
 	"basicauths.project.cattle.io":                                    false,
-	"catalogs.management.cattle.io":                                   false,
-	"catalogtemplates.management.cattle.io":                           false,
-	"catalogtemplateversions.management.cattle.io":                    false,
 	"certificates.project.cattle.io":                                  false,
 	"cloudcredentials.management.cattle.io":                           false,
 	"clusterauthtokens.cluster.cattle.io":                             false,
 	"clusterclasses.cluster.x-k8s.io":                                 false,
-	"clustermonitorgraphs.management.cattle.io":                       false,
+	"clusterloggings.management.cattle.io":                            false,
```
Comment (JonCrowther) [inline pkg/crds/names.go:186]: Remove `clusterloggings`

File: pkg/crds/names.go [hunk 10]
```diff
@@ -247,7 +231,7 @@ var MigratedResources = map[string]bool{
 	"podsecurityadmissionconfigurationtemplates.management.cattle.io": false,
 	"preferences.management.cattle.io":                                false,
 	"principals.management.cattle.io":                                 false,
-	"projectmonitorgraphs.management.cattle.io":                       false,
+	"projectloggings.management.cattle.io":                            false,
```
Comment (JonCrowther) [inline pkg/crds/names.go:234]: Remove `projectloggings`

File: pkg/data/management/role_data.go [hunk 11]
```diff
@@ -165,10 +160,8 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 		addRule().apiGroups("storage.k8s.io").resources("storageclasses").verbs("get", "list", "watch").
 		addRule().apiGroups("apiregistration.k8s.io").resources("apiservices").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("clusterevents").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("clustercatalogs").verbs("get", "list", "watch").
+		addRule().apiGroups("management.cattle.io").resources("clusterloggings").verbs("get", "list", "watch").
```
Comment (JonCrowther) [inline pkg/data/management/role_data.go:163]: Remove `clusterloggings`

File: pkg/data/management/role_data.go [hunk 12]
```diff
@@ -251,10 +238,7 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
 		addRule().apiGroups("").resources("persistentvolumeclaims").verbs("*").
 		addRule().apiGroups("metrics.k8s.io").resources("pods").verbs("*").
 		addRule().apiGroups("management.cattle.io").resources("clusterevents").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("clustercatalogs").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("projectcatalogs").verbs("*").
-		addRule().apiGroups("management.cattle.io").resources("catalogtemplates").verbs("*").
-		addRule().apiGroups("management.cattle.io").resources("catalogtemplateversions").verbs("*").
+		addRule().apiGroups("management.cattle.io").resources("projectloggings").verbs("*").
```
Comment (JonCrowther) [inline pkg/data/management/role_data.go:241]: Remove `projectloggings`

File: pkg/data/management/role_data.go [hunk 13]
```diff
@@ -272,19 +256,14 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
...
 		addRule().apiGroups("storage.k8s.io").resources("storageclasses").verbs("get", "list", "watch").
 		addRule().apiGroups("apiregistration.k8s.io").resources("apiservices").verbs("get", "list", "watch").
 		addRule().apiGroups("").resources("persistentvolumeclaims").verbs("*").
 		addRule().apiGroups("metrics.k8s.io").resources("pods").verbs("*").
 		addRule().apiGroups("management.cattle.io").resources("clusterevents").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("clustercatalogs").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("projectcatalogs").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("catalogtemplates").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("catalogtemplateversions").verbs("get", "list", "watch").
+		addRule().apiGroups("management.cattle.io").resources("projectloggings").verbs("get", "list", "watch").
```
Comment (JonCrowther) [inline pkg/data/management/role_data.go:266]: Remove `projectloggings`

File: pkg/data/management/role_data.go [hunk 14]
```diff
@@ -301,16 +280,13 @@ func addRoles(wrangler *wrangler.Context, management *config.ManagementContext)
...
-		addRule().apiGroups("project.cattle.io").resources("apprevisions").verbs("get", "list", "watch").
 		addRule().apiGroups("").resources("persistentvolumes").verbs("get", "list", "watch").
 		addRule().apiGroups("storage.k8s.io").resources("storageclasses").verbs("get", "list", "watch").
 		addRule().apiGroups("apiregistration.k8s.io").resources("apiservices").verbs("get", "list", "watch").
 		addRule().apiGroups("").resources("persistentvolumeclaims").verbs("get", "list", "watch").
 		addRule().apiGroups("metrics.k8s.io").resources("pods").verbs("get", "list", "watch").
 		addRule().apiGroups("management.cattle.io").resources("clusterevents").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("clustercatalogs").verbs("get", "list", "watch").
-		addRule().apiGroups("management.cattle.io").resources("projectcatalogs").verbs("get", "list", "watch").
+		addRule().apiGroups("management.cattle.io").resources("projectloggings").verbs("get", "list", "watch").
```
Comment (JonCrowther) [inline pkg/data/management/role_data.go:289]: Remove `projectloggings`

File: tests/integration/suite/test_role_template.py
Comment (JonCrowther) [inline tests/integration/suite/test_role_template.py:1]: This got re-added from here https://github.com/rancher/rancher/pull/47968

**Review by JonCrowther — COMMENTED** [review]
Looks like some changes were accidentally undone from the following PRs:
https://github.com/rancher/rancher/pull/47968
https://github.com/rancher/rancher/pull/47997
https://github.com/rancher/rancher/pull/42847

File: cmd/agent/main.go [hunk 15]
```diff
@@ -70,7 +70,6 @@ func main() {
 			err = errors.Join(
 				clean.DuplicateBindings(nil),
 				clean.OrphanBindings(nil),
-				clean.OrphanCatalogBindings(nil),
```
Comment (pjbgf) [inline cmd/agent/main.go:73]: Related code from `pkg/data/management/bindings_cleanup.go` seem to no longer be needed:

```golang
	orphanCatalogBindingsCleanupKey = "CleanupOrphanCatalogBindingsDone"

...

	err = cleanupSpecificOrphanedBindings(scaledContext, wContext, orphanCatalogBindingsCleanupKey)
	if err != nil {
		logrus.Errorf("failed to cleanup orphan catalog bindings")
	}
```

**Comment by pjbgf** [comment]
For upgrade paths, we need to ensure we don't have issues with:
- Existing bindings to the built-in roles we removed.
- Existing custom Roles/ClusterRoles that explicitly refer to the CRDs we are removing.

**Review by JonCrowther — DISMISSED** [review]

**Review by pjbgf — APPROVED** [review]
LGTM. Given its size, it may be worth merging it soon so it has more time to mature and get thoroughly tested before the release.

**Comment by JonCrowther** [comment]
> > For upgrade paths, we need to ensure we don't have issues with:
> > 
> > * Existing bindings to the built-in roles we removed.
> > * Existing custom Roles/ClusterRoles that explicitly refer to the CRDs we are removing.
> 
> The existings bindings to the built-in roles are automatically removed. Existing custom Roles/ClusterRoles are automatically updated. They dont contain rules about the CRDS we removed.

For any custom roles that reference these CRDs, if they don't get updated they won't cause errors. In k8s it's valid to have a Role or ClusterRole with resources that aren't installed. So it's not ideal UX, but it shouldn't cause any problems in a user's system if they aren't updated.

**Review by JonCrowther — APPROVED** [review]

File: tests/v2/codecoverage/package/Dockerfile [hunk 16]
```diff
@@ -41,19 +41,16 @@ WORKDIR /var/lib/rancher
...
 # kontainer-driver-metadata branch to be set for specific branch other than dev/master, logic at rancher/rancher/pkg/settings/setting.go
 ARG CATTLE_KDM_BRANCH=dev-v2.10
 
-ENV CATTLE_SYSTEM_CHART_DEFAULT_BRANCH=$SYSTEM_CHART_DEFAULT_BRANCH
 ENV CATTLE_CHART_DEFAULT_BRANCH=$CHART_DEFAULT_BRANCH
 ENV CATTLE_PARTNER_CHART_DEFAULT_BRANCH=$PARTNER_CHART_DEFAULT_BRANCH
 ENV CATTLE_RKE2_CHART_DEFAULT_BRANCH=$RKE2_CHART_DEFAULT_BRANCH
-ENV CATTLE_HELM_VERSION v2.16.8-rancher2
-ENV CATTLE_MACHINE_VERSION v0.15.0-rancher122
+ENV CATTLE_MACHINE_VERSION v0.15.0-rancher121
```
Comment (jiaqiluo) [inline tests/v2/codecoverage/package/Dockerfile:53]:
```suggestion
ENV CATTLE_MACHINE_VERSION v0.15.0-rancher122
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Comment by pjbgf** [comment]
> Just FYI, for any custom roles that reference these CRDs, if they don't get updated they won't cause errors. In k8s it's valid to have a Role or ClusterRole with resources that aren't installed. So it's not ideal UX, but it shouldn't cause any problems in a user's system if they aren't updated.

@JonCrowther thanks for clarifying this. Overall, that behaviour will come handy for users that may need to rollback into an older version.

**Review by jiaqiluo — DISMISSED** [review]
Good Job! 
LGTM 💯

**Review by pjbgf — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Review by JonCrowther — APPROVED** [review]

**Review by tashima42 — DISMISSED** [review]

**Review by diogoasouza — DISMISSED** [review]

**Review by pjbgf — APPROVED** [review]
@rohitsakala great job getting this over the line! :1st_place_medal: :bow:

**Review by prachidamle — APPROVED** [review]

**Review by diogoasouza — APPROVED** [review]
